tesseract-server/src/logic_layer/cache.rs

```diff
@@ -0,0 +1,196 @@
+use failure::{Error, format_err};
+use log::*;
+
+use tesseract_core::{Schema, Cube, Dimension, Backend, ColumnData};
+
+
+/// Holds cache information.
+#[derive(Debug, Clone)]
+pub struct Cache {
+    pub cubes: Vec<CubeCache>,
+}
+
+impl Cache {
+    /// Finds the `CubeCache` object for a cube with a given name.
+    pub fn find_cube_info(&self, cube: &String) -> Option<CubeCache> {
+        for cube_cache in &self.cubes {
+            if cube_cache.name == *cube {
+                return Some(cube_cache.clone());
+            }
+        }
+        None
+    }
+}
+
+/// Holds cache information for a given cube.
+#[derive(Debug, Clone)]
+pub struct CubeCache {
+    pub name: String,
+    pub time_dim: Dimension,
+    pub years: Vec<i16>,
+}
+
+impl CubeCache {
+    /// Returns dimension name in the format: `Dimension.Hierarchy.Level`.
+    pub fn get_time_dim_name(&self) -> String {
+        format!("{}.{}.{}",
+            self.time_dim.name,
+            self.time_dim.hierarchies[0].name,
+            self.time_dim.hierarchies[0].levels[0].name,
+        ).to_string()
+    }
+
+    pub fn get_year_cut(&self, s: String) -> Result<String, Error> {
```

Comment:
it's mentioned a little above. But I think it would be nice to have a `Year` enum, with `Latest` and `First` and `Value(usize)` so that it's really clear in the code what the allowed values are.